### ISSUE TITLE
Support direct logging of SF HTTP error responses

### ISSUE THREAD

**Issue by arbokrad** [jongpie/NebulaLogger#299]
### New Feature Summary

Considering the following code sample:

```
someFunction() {
	const logger = this.template.querySelector('c-logger');

	try {	
		someCallToApex()
			.then( result => {
				// success path
			})
			.catch( error => {
				// currently, logger.error(...) expects a JS error/exception
				// but this is not the signature of the error coming back from SF's calls
                                logger.error( error );
			});
	} finally {
		logger.saveLog();
	}
}
```

...it would be great to be able to take an error http response and directly log it using the logger

Proposing: `logger.errorHttp(...)` or something to that effect

**Comment by jongpie** [comment]
@arbokrad thanks for adding this issue! I think adding support for logging these types of errors makes a lot of sense - I might end up keeping the method as just `logger.error()` (instead of `logger.errorHttp(...)`) and have the `error()` method support both types of parameters, but I'll do some prototyping to see if there are any issues with that approach.

I'm not 100% sure when I'll implement this, but my hope is sometime within the next month or two (I'll probably implement it in the same release as #240).

**Comment by jongpie** [comment]
@arbokrad I think I've just about got this change ready to release! I've updated the logic in the JS function `setError()` to handle either a JavaScript Error (existing functionality) or an Apex error (new), so you'll be able to log Apex HTTP errors the same way you log JavaScript errors. Here's a screenshot of an example - the green box shows that it still logs the LWC as the origin (in this case, I'm using a LWC called `loggerLWCDemo`), but the exception info (red box) shows that it was an Apex error (that originated in an Apex class called `LoggerLWCDemoController`). 

![image](https://user-images.githubusercontent.com/1267157/171646616-83dda0bf-df62-4553-b261-b8ef8b4a1b78.png)

I want to wrap up a little more testing & code cleanup, but I'm hoping to get this released as `v4.7.6` in the next day or so.

**Comment by jongpie** [comment]
@arbokrad it's finally ready! I've just released this functionality in [v4.7.6](https://github.com/jongpie/NebulaLogger/releases/tag/v4.7.6) - whenever you have a chance to check it out, let me know if you have any feedback.
